src/elements/content-sidebar/versions/VersionsSidebar.js

```diff
@@ -27,22 +29,59 @@ type Props = {
     isLoading: boolean,
     onUpgradeClick?: () => void,
     parentName: string,
+    showUpsell: boolean,
+    showUpsellWithPicture: boolean,
     versionCount: number,
     versionLimit: number,
     versions: Array<BoxItemVersion>,
 };
 
-const VersionsSidebar = ({ error, errorTitle, onUpgradeClick, isLoading, parentName, versions, ...rest }: Props) => {
+const VersionsSidebar = ({
+    error,
+    errorTitle,
+    onUpgradeClick,
+    isLoading,
+    parentName,
+    showUpsell,
+    showUpsellWithPicture,
+    versions,
+    ...rest
+}: Props) => {
     const showLimit = versions.length >= MAX_VERSIONS;
     const showVersions = !!versions.length;
     const showEmpty = !isLoading && !showVersions && !onUpgradeClick;
     const showError = !!error;
 
+    const upsellOverlay = () => {
+        if (showUpsell || showUpsellWithPicture) {
```

Comment:
Can we move this to a guard clause and `return null` early if either is not true?